src/main/kotlin/io/snyk/plugin/extensions/SnykControllerLifecycleManager.kt

```diff
@@ -0,0 +1,11 @@
+package io.snyk.plugin.extensions
+
+// SnykControllerLifecycleManager is the extension point interface
+// which other plugins can implement in order to integrate with Snyk.
+interface SnykControllerLifecycleManager {
+
+    // register is called by Snyk on third-party implementations of this interface
```

Comment:
I'd make this more explicit and a JavaDoc comment (`/** */`). Also, while I'm nitpicking ;), I'd remove `Lifecycle` from the name. How about: `register is called by the Snyk IntelliJ plugin to pass the #SnykController to extension point implementers`.